sdk/typescript/src/transactions/Transaction.ts

```diff
@@ -241,7 +275,10 @@ export class Transaction {
 	}
 
 	constructor() {
+		const globalPlugins = getGlobalPluginRegistry();
 		this.#data = new TransactionDataBuilder();
+		this.#buildPlugins = [...globalPlugins.buildPlugins];
```

Comment:
This is fine to me but I do wonder if we should lazy resolve these via getters instead of during transaction construction.

Reply:
this is a good question... I went back and forth on this.  I went with this way because I wasn't sure if it would be expected that transactions get new plugins after they are created.

Specifically, you could build a transaction multiple times and have it run a different set of plugins if they are added after between builds, which seemed a bit off.  But I can also see why that might be desirable.

Reply:
Yeah I buy that argument. We can always change this at some point if it ends up being unintuitive anyway